File: Backend/application_service/core/domain/competition.go
package domain

import (
	"github.com/google/uuid"
	"time"
)

type Competition struct {
	ID                            uuid.UUID
	Name                          string
	StartDate                     time.Time
	EndDate                       time.Time
	Gender                        Gender
	Type                          CompetitionType
	Tiebreak                      bool
	AddressId                     uuid.UUID
	Address                       Address
	OrganizerID                   uuid.UUID
	Organizer                     SportsOrganization
	DelegationMemberPropositionID uuid.UUID
	DelegationMemberProposition   DelegationMemberProposition
	TeamCompositionID             uuid.UUID
	TeamComposition               TeamComposition
	AgeCategories                 []AgeCategory
}

File: Frontend/Public/gco-public/src/app/model/core/competition.ts
import {Gender} from './gender'
import {Address} from './address'
import {SportsOrg} from './sports-org'
import { CompetitionType } from './competition-type';
import { DelegationMemberProposition } from './delegation-member-proposition';
import { TeamComposition } from './team-composition';
import { AgeCategory } from './age-category'

export interface Competition{
    id?: string;
    name: string;
    startDate: number;
    endDate: number;
    gender: Gender;
    type: CompetitionType;
    tiebreak: boolean;
    address: Address;
    organizer?: SportsOrg;

    delegationMemberPropositions?: DelegationMemberProposition[];
    teamComposition: TeamComposition;
    ageCategories?: AgeCategory[];
}


export interface CompetitionTable {
    displayedColumns: string[];
    dataSource: Competition[];
    selectedRow: Competition | null;
}



